Formula/d/dwarfs.rb

```diff
@@ -0,0 +1,148 @@
+class Dwarfs < Formula
+  desc "Fast high compression read-only file system for Linux, Windows, and macOS"
+  homepage "https://github.com/mhx/dwarfs"
+  url "https://github.com/mhx/dwarfs/releases/download/v0.10.1/dwarfs-0.10.1.tar.xz"
+  sha256 "db785e0e0f257fa4363d90153db34127add4552791a72998b30ded787840d039"
+  license "GPL-3.0-or-later"
+
+  depends_on "cmake" => :build
+  depends_on "googletest" => :build
+  depends_on "pkg-config" => :build
+  depends_on "boost"
+  depends_on "brotli"
+  depends_on "double-conversion"
+  depends_on "flac"
+  depends_on "fmt"
+  depends_on "gflags"
+  depends_on "glog"
+  depends_on "howard-hinnant-date"
+  depends_on "libarchive"
+  depends_on "libevent"
+  depends_on "libsodium"
+  depends_on "llvm" if OS.mac? && DevelopmentTools.clang_build_version < 1500
+  depends_on "lz4"
+  depends_on "nlohmann-json"
+  depends_on "openssl@3"
+  depends_on "parallel-hashmap"
+  depends_on "range-v3"
+  depends_on "utf8cpp"
+  depends_on "xxhash"
+  depends_on "xz"
+  depends_on "zstd"
+
+  on_linux do
+    depends_on "jemalloc"
+  end
+
+  fails_with :clang do
+    build 1400
+    cause "Not all required C++20 features are supported"
+  end
+
+  fails_with :gcc do
+    version "11"
+    cause "Not all required C++20 features are supported"
+  end
+
+  def install
+    args = std_cmake_args + %W[
+      -DBUILD_SHARED_LIBS=ON
+      -DCMAKE_PREFIX_PATH=#{libexec}
+      -DCMAKE_INSTALL_RPATH=#{rpath}
+      -DWITH_LIBDWARFS=ON
+      -DWITH_TOOLS=ON
+      -DWITH_FUSE_DRIVER=OFF
+      -DWITH_TESTS=ON
+      -DWITH_MAN_PAGES=ON
+      -DENABLE_PERFMON=ON
+      -DTRY_ENABLE_FLAC=ON
+      -DENABLE_RICEPP=ON
+      -DENABLE_STACKTRACE=OFF
+      -DDISABLE_MOLD=ON
+      -DPREFER_SYSTEM_GTEST=ON
+    ]
+
+    if OS.mac? && DevelopmentTools.clang_build_version < 1500
+      ENV.llvm_clang
+
+      # Needed in order to find the C++ standard library
+      # See: https://github.com/Homebrew/homebrew-core/issues/178435
+      ENV.prepend "LDFLAGS", "-L#{Formula["llvm"].opt_lib}/c++ -L#{Formula["llvm"].opt_lib}"
+    end
+
+    system "cmake", "--fresh", "-S", ".", "-B", "build", *args
+    system "cmake", "--build", "build", "--parallel", ENV.make_jobs
```

Comment:
Let's have CMake determine its own thread limit.
```suggestion
    system "cmake", "--build", "build"
```

Reply:
Without `--parallel` (or `-J`), it'll always run only a single process.

With *just* `--parallel` instead of `--parallel` + `ENV.make_jobs`, it'll try to derive this from the CPU count.

Removing this completely is most definitely a bad idea. The `ENV.make_jobs` part is used by various other formulae, so I was just following precedent here.

Reply:
> The ENV.make_jobs part is used by various other formulae, so I was just following precedent here.

Only CMake itself uses this for a CMake build though.
https://cmake.org/cmake/help/v3.22/manual/cmake.1.html#build-a-project also suggests that it will in fact build in parrallel, and the option is only to set an upper limit.

Reply:
I've removed the `ENV.make_jobs` part.